cmd/telegraf/telegraf.go

```diff
@@ -151,6 +155,40 @@ func reloadLoop(
 	}
 }
 
+// configDirFiles returns a slice made up of filepaths correspoding
+// to config files defined in any of the config directories.
+func configDirFiles() sliceFlags {
+	configs := make(sliceFlags, 0)
+	for _, dir := range fConfigDirs {
+		walkfn := func(thispath string, info os.FileInfo, _ error) error {
+			if info == nil {
+				log.Printf("W! Telegraf is not permitted to read %s", thispath)
+				return nil
+			}
+
+			if info.IsDir() {
+				// skip Kubernetes mounts, preventing loading the same config twice
```

Comment:
I didn't remove this check. I don't immediately know if it also applies in this context, so I have to look a bit more. If it is necessary, the comment should probably be updated anyway. Happy for you to weigh in.